django/db/models/query.py

```diff
@@ -1184,6 +1184,48 @@ def _raw_delete(self, using):
 
     _raw_delete.alters_data = True
 
+    def bulk_delete(self):
+        """
+        Delete the records in the current QuerySet with more efficiency.
+
+        This method performs a fast deletion for all the models in the QuerySet,
+        ignoring all signal calls and optimizing the deletion process for models
+        with no foreign key references and inheritance.
+
+        For models with foreign key refrences and inheritance, it performs same
+        as the normal Queryset.delete() method.
+        """
+        self._not_support_combined_queries("bulk_delete")
+        if self.query.is_sliced:
+            raise TypeError("Cannot use 'limit' or 'offset' with bulk_delete().")
+        if self.query.distinct or self.query.distinct_fields:
+            raise TypeError("Cannot call bulk_delete() after .distinct().")
+        if self._fields is not None:
+            raise TypeError("Cannot call bulk_delete() after values() or values_list()")
+
+        del_query = self._chain()
+
+        # The delete is actually 2 queries - one to find related objects,
+        # and one to delete. Make sure that the discovery of related
+        # objects is performed on the same database as the deletion.
+        del_query._for_write = True
+
+        # Disable non-supported fields.
+        del_query.query.select_for_update = False
+        del_query.query.select_related = False
+        del_query.query.clear_ordering(force=True)
+
+        collector = Collector(using=del_query.db, origin=self, perform_signals=False)
+        collector.collect(del_query)
+        deleted, _rows_count = collector.delete()
+
+        # Clear the result cache, in case this QuerySet gets reused.
+        self._result_cache = None
+        return deleted, _rows_count
+
+    bulk_delete.alters_data = True
+    bulk_delete.queryset_only = True
```

Comment:
Given the implementation of this method is identical to that of `delete`, except for setting `perform_signals=False` in `Collector`, I think we should either merge the two methods or delegate the implementation of each to a new private method (`_delete` perhaps).